spec/mongoid/scroll_cursor_spec.rb

```diff
@@ -138,4 +138,24 @@
       end.to raise_error ArgumentError
     end
   end
+  context 'a cursor with include_current set to true' do
```

Comment:
Tell me if you think that it should be tested for each datatype or be written differently.

Reply:
I don't think it's necessary. What you should have is tests that scroll with vs. without `include_current` set to `true`, and check the actual data returned. Basically the example in your PR that describes what happens when. And since this may need special care with a tiebreak, add a test for that.